firefox-ios/Client/Frontend/Browser/TabScrollController.swift

```diff
@@ -174,20 +178,19 @@ class TabScrollingController: NSObject,
                                                selector: #selector(applicationWillTerminate(_:)),
                                                name: UIApplication.willTerminateNotification,
                                                object: nil)
+        // need to add this manually otherwise listenForThemeChanges(view) will retain the view in memory
+        // causing memory leaks
+        themeObserver = notificationCenter.addObserver(name: .ThemeDidChange, queue: .main) { [weak self] _ in
+            self?.applyTheme()
+        }
     }
 
-    private func setupTabOnLoadingCallback() {
-        if isPullToRefreshRefactorEnabled {
-            tab?.onLoading = { [weak self] in
-                guard let self else { return }
-                // If we are in the home page delete pull to refresh so it doesn't show on the background
-                if tabIsLoading() || (tab?.isFxHomeTab ?? false) {
-                    pullToRefreshView?.stopObservingContentScroll()
-                    pullToRefreshView?.removeFromSuperview()
-                } else {
-                    configureRefreshControl(isEnabled: true)
-                }
-            }
+    private func onTabLoading() {
```

Comment:
I think this function name doesn't really tell me what this code is doing. Is there a method name that would potentially be more descriptive of what is happening here?

Reply:
Yes i agree i'm modifying it